helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -336,7 +342,32 @@ public Integer apply(final Response reply) {
   }
 
   public ListenableFuture<List<String>> listHosts() {
-    return get(uri("/hosts/"), new TypeReference<List<String>>() {
+    return listHosts(ImmutableMultimap.<String, String>of());
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern) {
+    return listHosts(ImmutableMultimap.of("namePattern", namePattern));
+  }
+
+  public ListenableFuture<List<String>> listHosts(final List<String> labels) {
+    final Multimap<String, String> query = HashMultimap.create();
+    query.putAll("labels", labels);
+
+    return listHosts(query);
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern,
+                                                  final List<String> labels) {
```

Comment:
I would prefer to leave it as a List of String so that the HeliosClient user can pass in any form of selector without having to necessarily have the operator defined in it's code - since HostSelector and the parsing is done in a class included in HeliosClient. This way if there are new operators it won't require a new HeliosClient version.

We could though add overloads of this method for both parsed and unparsed parameters if anyone finds that useful

```java
listHost(List<String> unparsedSelectors);
listHost(List<HostSelector> selectors);
```